tests/unit/test_compat.py

```diff
@@ -40,6 +41,29 @@ def test_get_path_uid_symlink_without_NOFOLLOW(tmpdir, monkeypatch):
         get_path_uid(fs)
 
 
+def test_console_to_str(monkeypatch):
+    some_bytes = b"a\xE9\xC3\xE9b"
+    encodings = ('ascii', 'utf-8', 'iso-8859-1', 'iso-8859-5',
+                 'koi8_r', 'cp850')
+    for e in encodings:
+        monkeypatch.setattr(locale, 'getpreferredencoding', lambda: e)
+        result = console_to_str(some_bytes)
+        assert result.startswith("a")
+        assert result.endswith("b")
+
+
+def test_console_to_str_warning(monkeypatch):
+    some_bytes = b"a\xE9b"
+
+    def check_warning(msg):
+        assert msg.startswith(
+            "Subprocess output does not appear to be encoded as")
+
+    monkeypatch.setattr(locale, 'getpreferredencoding', lambda: 'utf-8')
+    monkeypatch.setattr(pip.compat.logger, 'warning', check_warning)
+    console_to_str(some_bytes)
```

Comment:
Maybe it should be asserted that this monkey patched function is called?

Reply:
Could do I guess. The idea of the monkeypatching is to fix the behaviour rather than to test those functions are called, but it wouldn't do any harm.